Userland/Utilities/shot.cpp

```diff
@@ -5,28 +5,100 @@
  */
 
 #include <AK/Format.h>
+#include <AK/Optional.h>
 #include <LibCore/ArgsParser.h>
 #include <LibCore/DateTime.h>
 #include <LibCore/File.h>
 #include <LibGUI/Application.h>
 #include <LibGUI/Clipboard.h>
+#include <LibGUI/Painter.h>
+#include <LibGUI/Widget.h>
+#include <LibGUI/Window.h>
 #include <LibGUI/WindowServerConnection.h>
 #include <LibGfx/PNGWriter.h>
 #include <unistd.h>
 
+class SelectableLayover final : public GUI::Widget {
```

Comment:
It feels rather awkward to do click-click or click-click+drag to select a region, why not just have a normal region select mechanism similar to pixelpaint? i.e. click+drag to select, and release to take a screenshot?

Reply:
Switched to click+drag to select, the code is also much cleaner now :)